c/src/ml-api-service-offloading.h

```diff
@@ -14,17 +14,61 @@
 #define __ML_SERVICE_OFFLOADING_INTERNAL_H__
 
 #include <ml-api-service.h>
+#include <nnstreamer-edge.h>
+
 #include "nnstreamer-tizen-internal.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif /* __cplusplus */
 typedef void * ml_service_offloading_h;
 
+/**
+ * @brief Enumeration for ml-offloading type.
+ */
+typedef enum
+{
+  ML_OFFLOADING_TYPE_UNKNOWN = 0,
+  ML_OFFLOADING_TYPE_TRAINING,
+
+  ML_OFFLOADING_TYPE_MAX
+} ml_offloading_type_e;
+
+/*/
+ * @brief Structure for ml_service_offloading
+ */
+typedef struct
+{
+  nns_edge_h edge_h;
+  nns_edge_node_type_e node_type;
+
+  gchar *path; /**< A path to save the received model file */
+  ml_option_h info;
+  GHashTable *table;
+
+  ml_offloading_type_e offloading_type;
+  void *priv;
+} _ml_service_offloading_s;
+
 /**
  * @brief Creates ml offloading service handle with given ml-option handle.
  * @remarks The @a handle should be destroyed using ml_service_destroy().
  * @param[in] handle ml-service handle created by ml_service_new().
+ * @return @c 0 on Success. Otherwise a negative error value.
+ * @retval #ML_ERROR_NONE Successful.
+ * @retval #ML_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #ML_ERROR_INVALID_PARAMETER Fail. The parameter is invalid.
+ * @retval #ML_ERROR_OUT_OF_MEMORY Failed to allocate required memory.
+ * @retval #ML_ERROR_STREAMS_PIPE Failed to launch the pipeline.
+ * @retval #ML_ERROR_TRY_AGAIN The pipeline is not ready yet.
+ * @retval #ML_ERROR_PERMISSION_DENIED The application does not have the privilege to access to the storage.
+ */
+int ml_service_offloading_create (ml_service_h handle);
```

Comment:
Do you intend to make this a public API?

Reply:
No, this is removed, thank you.